libraries/lib-sentry-reporting/SentryReport.cpp

```diff
@@ -243,12 +257,12 @@ Exception& Exception::SetHandled(bool Handled)
 
 Message Message::Create(std::string message)
 {
-   return { std::move(message) };
+   return { AnonymizePaths(std::move(message)) };
 }
 
 Message& Message::AddParam(std::string value)
 {
-   Params.emplace_back(std::move(value));
+   Params.emplace_back(AnonymizePaths(std::move(value)));
```

Comment:
I suggest:

AnonymizedString is a new class that wraps a const std::string.  It calls the function in its constructor.  It has operator const std::string& .  It has a method called `move()` that returns std::string&& .

So objects of that type guarantee anonymization at construction, and are immutable except to move out of them again.

Then Message::Create and all these other functions are changed to take AnonymizedString as argument.

Why would I do this?  Two reasons.  One, I genrally don't like "naked" wxString or other string classes overused in function signatures.  I would rather always have strings in different wrapper classes when they serve different purposes, and the non-interconversion of those wrapper types eliminates programming errors.

Second reason, it makes it more obvious to anyone reviewing classes Message and Exception, just from the header files, that they deal in anonymized data:  It's not implementation detail inside those functions, that anonymization happens.  Rather the reviewer is directed to the AnonymizedString class and can see it must happen before you enter these functions.

Reply:
>when they serve different purposes

Well, they are just strings here. But I like your second point

Reply:
A string is not just a string if you wrap it in a well designed class, with a constructor that establishes an important invariant property (like some kind of well-formedness), and methods that preserve this invariant (such as by making the contents immutable or only available to move out).

A function taking that wrapped class is saying in its contract, not its implementation, that it takes as a precondition the prior satisfaction of this invariant.  Successful construction of an object of the class amounts to a proof of the precondition.

Maybe I am only restating my second point here, but I find it very useful to think in such terms.

Reply:
I use `@pre` `@post` and `@invariant` in my Doxygen comments sometimes.  Programming by contract, even if only informally stated in comments, is useful discipline.